python/mlir_structured/runtime/util.py

```diff
@@ -24,3 +27,29 @@ def mlir_mod_ctx(src: Optional[str] = None,
       module = Module.create()
     with InsertionPoint(module.body):
       yield module
+
+
+def print_check_string(op,
+                       replacements,
+                       next_line="",
+                       comment_sigil="#",
+                       generic_op_form=False):
+  s = op.get_asm(print_generic_op_form=generic_op_form)
+  for k, v in replacements.items():
+    s = s.replace(k, f"%[[{v}]]")
+
+  s = re.sub(r"%\w+", "%{{.*}}", s)
+  previous_frame = inspect.currentframe().f_back
+  (filename, line_number, function_name, lines,
+   index) = inspect.getframeinfo(previous_frame)
+  with open(filename) as source_file:
+    source_file_lines = source_file.readlines()
+  previous_line = source_file_lines[line_number - 2]
+  leading_white_space = len(previous_line) - len(previous_line.strip(' '))
+  leading_white_space = previous_line[:leading_white_space]
+
+  print()
+  print(previous_line.rstrip())
+  print(indent(s, leading_white_space + f"{comment_sigil} CHECK: "))
+  print(leading_white_space + next_line)
```

Comment:
This is useful little function but it's for testing purposes only so I'm not sure where to put it.